### DISCUSSION TITLE
Error allocating 32gb of memory??

### DISCUSSION THREAD

**Discussion by OliverGeneser** [apache/couchdb#4571] [Q&A]
Hey! :smiley: 

I'm having a problem with running a old CouchDB project in docker on my fedora 37/38 machines.

I'm getting a memory allocations error because it's trying to allocate 32GB but there is only 16GB available...

It's an old CouchDB version (1.7) but it works on other machines with 16GB

Error:
```
server-db-couch-1  | ll_alloc: Cannot allocate 34359738112 bytes of memory (of type "fd_tab", thread 0).
server-db-couch-1  | /bin/bash: line 1:     9 Aborted                 (core dumped) couchdb
server-db-couch-1 exited with code 134
```

**Comment by sroelants** [comment]
I was running into this exact same issue. I won't pretend to understand exactly what the underlying issue is. It looks to be a problem on the Erlang-end, where it's trying to allocate enough memory (`ll_alloc` allocates long-lived memory) to store its file descriptor table (memory of type `fd_tab`).

Originally I worked around it by adding a large enough swap file, so my system would have enough memory available to accommodate for this outrageous requirement, but the couchdb performance suffered dramatically.

The only thing that seemingly works so far is limiting the number of allowed open file handles at the docker level, like suggested here: https://erlang.org/pipermail/erlang-questions/2016-May/089335.html

Try running `docker run` with an additional `--ulimit nofile=<SOFT_LIMIT>:<HARD_LIMIT>` flag. Or, if you're using `docker compose`, [try adding the `ulimits` there](https://docs.docker.com/compose/compose-file/05-services/#ulimits):
```
services:
  couchdb:
    image: couchdb:1.7
    ulimits:
      nofile:
        soft: <SOFT>
        hard: <HARD>
```
